compiler/rustc_parse/src/parser/item.rs

```diff
@@ -2385,7 +2385,22 @@ impl<'a> Parser<'a> {
             };
             let (pat, ty) = if is_name_required || this.is_named_param() {
                 debug!("parse_param_general parse_pat (is_name_required:{})", is_name_required);
-
+                if let token::Ident(symname, _) = this.token.kind
+                    && symname.as_str().eq("Sized")
+                    && this.prev_token.is_lifetime()
+                {
+                    // parse pattern with "'a Sized" we're supposed to give suggestion like
+                    // "'a + Sized"
+                    let mut err = this.struct_span_err(this.prev_token.span, "expected `+` between lifetime and trait");
+                    err.multipart_suggestion(
+                    "missing `+`",
```

Comment:
This should be an instruction for the user: "add `+`"

Reply:
fixed